### PR TITLE
Add license and publish

### PR DESCRIPTION
**Pull request by dio**
This upgrades docusaurus to 2.0.0-beta.15, add `make license` (to [addlicense](https://github.com/google/addlicense) to files without license), also publish the site to gh-pages.

Signed-off-by: Dhi Aurrahman <dio@rockybars.com>

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -45,3 +45,8 @@ jobs:
           key: protosaurus-${{ runner.os }}-buf-${{ hashFiles('testdata/buf.lock') }}
           restore-keys: protosaurus-${{ runner.os }}-buf-
       - run: make gen
+      - run: make docs
+      - uses: peaceiris/actions-gh-pages@v3
```
Comment (dio) [inline .github/workflows/ci.yaml:53]: Will guard this when only on `main`.

Comment (dio) [reply .github/workflows/ci.yaml:53]: Guarded.

**Comment by dio** [comment]
Not sure if we need to set anything to make the `gh-pages` to work.

**Comment by dio** [comment]
1. Go to the setting of this repository
2. In the "Github Pages" section make sure to have "gh-pages" selected under "Source"

File: Makefile [hunk 2]
```diff
@@ -78,6 +79,14 @@ gen: $(BUF_V1_MODULE_DATA) $(yarn) ## Generate files from proto files
 format: $(buf) $(clang-format) ## Format all proto files
 	@$(clang-format) -i $(shell $(buf) ls-files)
 
+docs: $(yarn) ## Build the docs site
+	$(yarn) --cwd website
+	$(yarn) --cwd website build
+
+license_files := website/plugins testdata .github Makefile *.mk buf.*.yaml
```
Comment (imballinst) [inline Makefile]: How about the .css and .js files under `website/src`?

**Comment by imballinst** [comment]
> 1. Go to the setting of this repository
> 
>     2. In the "Github Pages" section make sure to have "gh-pages" selected under "Source"

Done! Thanks for setting up the CD.

**Comment by imballinst** [comment]
Probably we need to update the `baseUrl` as well: 

https://github.com/imballinst/protosaurus/blob/2f7df68370a109de7b240e0cb9f9e6675f73850f/website/docusaurus.config.js#L14

```
if GHA environment then "/docusaurus" else "/"
```

**Comment by dio** [comment]
@imballinst I think we need to re-enable the `gh-pages`? I deleted the remote branch once 😅. Sorry.

**Comment by imballinst** [comment]
> @imballinst I think we need to re-enable the `gh-pages`? I deleted the remote branch once sweat_smile. Sorry.

done! All seems to be working normally here: https://imballinst.github.io/protosaurus/. Nice! :sparkles:

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -45,3 +49,12 @@ jobs:
...
+        if: ${{ github.ref == 'refs/heads/main' }}
+        with:
+          github_token: ${{ github.token }}
+          publish_dir: website/build
+          # This allows you to make your publish branch with only the latest commit.
+          # https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-force-orphan-force_orphan.
+          force_orphan: true
```
Comment (imballinst) [inline .github/workflows/ci.yaml:60]: This is brilliant! TIL about this GitHub Action.

**Review by imballinst — APPROVED** [review]
LGTM! Thanks for working on this!

**Pull request merged**